rust/lance-core/src/io/object_store.rs

```diff
@@ -317,6 +319,19 @@ impl ObjectStoreParams {
     pub fn set_commit_lock<T: CommitLock + Send + Sync + 'static>(&mut self, lock: Arc<T>) {
         self.commit_handler = Some(Arc::new(lock));
     }
+
+    /// Create a new instance of [`ObjectStoreParams`] based on the AWS credentials.
+    pub fn with_aws_credentials(
+        aws_credentials: Option<AwsCredentialProvider>,
+        region: Option<String>,
+    ) -> Self {
```

Comment:
This was previously defined (https://docs.rs/lance/0.8.14/lance/io/object_store/struct.ObjectStoreParams.html#method.with_aws_credentials) and was used in LanceDB. Bringing this back to keep things easy.